etc/service-bus.api.ts

```diff
@@ -0,0 +1,411 @@
+  update(settled: boolean, state?: any): void;
+}
+
+// @public
+interface Dictionary<T> {
+  // (undocumented)
+  [key: string]: T;
+}
+
+// @public (undocumented)
+interface MessageHandlerOptions {
+  autoComplete?: boolean;
+  maxAutoRenewDurationInSeconds?: number;
+  maxConcurrentCalls?: number;
+}
+
+// @public
+interface MessageHeader {
+  delivery_count?: number;
+  durable?: boolean;
+  first_acquirer?: boolean;
+  priority?: number;
+  ttl?: number;
+}
+
+// @public
+interface MessageProperties {
+  absolute_expiry_time?: number;
+  content_encoding?: string;
+  content_type?: string;
+  correlation_id?: string | number | Buffer;
+  creation_time?: number;
+  group_id?: string;
+  group_sequence?: number;
+  message_id?: string | number | Buffer;
+  reply_to?: string;
+  reply_to_group_id?: string;
+  subject?: string;
+  to?: string;
+  user_id?: string;
+}
+
+// @public
+class MessageSession extends LinkEntity {
+  // WARNING: The type "ClientEntityContext" needs to be exported by the package (e.g. added to index.ts)
+  // WARNING: The type "MessageSessionOptions" needs to be exported by the package (e.g. added to index.ts)
+  constructor(context: ClientEntityContext, options?: MessageSessionOptions);
+  autoComplete: boolean;
+  autoRenewLock: boolean;
+  close(): Promise<void>;
+  // WARNING: The type "ClientEntityContext" needs to be exported by the package (e.g. added to index.ts)
+  // WARNING: The type "MessageSessionOptions" needs to be exported by the package (e.g. added to index.ts)
+  static create(context: ClientEntityContext, options?: MessageSessionOptions): Promise<MessageSession>;
+  getState(): Promise<void>;
+  isOpen(): boolean;
+  maxAutoRenewDurationInSeconds: number;
+  maxConcurrentCallsPerSession?: number;
+  maxConcurrentSessions?: number;
+  maxMessageWaitTimeoutInSeconds: number;
+  peek(messageCount?: number): Promise<ReceivedMessageInfo[]>;
+  peekBySequenceNumber(fromSequenceNumber: Long, messageCount?: number): Promise<ReceivedMessageInfo[]>;
+  // WARNING: The type "SessionHandlerOptions" needs to be exported by the package (e.g. added to index.ts)
+  receive(onSessionMessage: OnSessionMessage, onError: OnError, options?: SessionHandlerOptions): void;
+  receiveBatch(maxMessageCount: number, maxWaitTimeInSeconds?: number): Promise<ServiceBusMessage[]>;
+  receiveDeferredMessage(sequenceNumber: Long): Promise<ServiceBusMessage | undefined>;
+  receiveDeferredMessages(sequenceNumbers: Long[]): Promise<ServiceBusMessage[]>;
+  receiveMode: ReceiveMode;
+  renewLock(): Promise<Date>;
+  sessionId?: string;
+  sessionLockedUntilUtc?: Date;
+  setState(state: any): Promise<void>;
+  // WARNING: The type "DispositionType" needs to be exported by the package (e.g. added to index.ts)
+  // WARNING: The type "DispositionOptions" needs to be exported by the package (e.g. added to index.ts)
+  settleMessage(message: ServiceBusMessage, operation: DispositionType, options?: DispositionOptions): Promise<any>;
```

Comment:
settleMessage is intended to be used internally. TypeScript doesn't support declaring an internal access modifier like C#. It only supports public, private, protected. Hence the problem arises. settleMessage() is called from ServiceBusMessage class when the user calls .complete(), .abandon(), .defer() or .deadLetter().

Reply:
Do we have a solution for these methods?